### PR TITLE
Move PersistentCache to flow/

### PR DESCRIPTION
**Pull request by arbreng**
PersistentCache no longer manages its version internally.  It is the
client's responsibility to inject a version directory at its discretion.

shell/gpu/ no longer depends on shell/common after this, so there is no
need to permute it for product / non-product configurations.

BUG: fxb/52961

### CODE REVIEW

File: shell/gpu/gpu_surface_gl.cc [hunk 1]
```diff
@@ -2,13 +2,13 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "gpu_surface_gl.h"
```
Comment (liyuqian) [inline shell/gpu/gpu_surface_gl.cc:5]: Is the change in line 5 necessary? I don't see the style guide of it on https://google.github.io/styleguide/cppguide.html

Comment (arbreng) [reply shell/gpu/gpu_surface_gl.cc:5]: Agreed, I'm not sure why I made this change back then, and I've removed it from the latet version

**Comment by arbreng** [comment]
> Moving PersistentCache into flow looks good as a follow-up step of #18938.
> 
> However, I'm a little concerned on the change of `SetCacheVersion`. It seems we didn't call `SetCacheVersion` anywhere in this PR, or set its default value. Does that mean we're just using an empty string as the version, so upgrading the engine version won't remove the old cache and create a new folder?

The latest PR sets the version properly from the Shell now.  It was a WIP before.

> It's also unclear why we no longer want to set the engine version as the cache version, and instead let the embedder set it. The fxb/52961 seems to be only about build configs, and there's no mention of persistent cache versions.

It's not strictly necessary to have the embedder set the version, but I did that to preserve a good layering between shell/ and flow/ (shell depends on flow, not the other way around).   This also allows the unittests to set their own version, and they are not tied to any hard-coded versioning logic like the Shell has.

**Comment by arbreng** [comment]
Needs minor cleanup and rebasing on https://github.com/flutter/engine/pull/20422

**Pull request closed without merging**